### PR TITLE
Fix service-catalog serviceclass-viewer role

### PR DESCRIPTION
**Pull request by pmorie**
Fixes the serviceclass-viewer role to correctly work with the `system:authenticated` group.

### CODE REVIEW

File: examples/service-catalog/service-catalog-rbac.yaml [hunk 1]
```diff
@@ -26,8 +26,9 @@ objects:
   roleRef:
     kind: ClusterRole
     name: servicecatalog-serviceclass-viewer
-  groupNames:
-  - system:authenticated
+  subjects:
+  - kind: Group
```
Comment (csrwng) [inline examples/service-catalog/service-catalog-rbac.yaml]: s/Group/SystemGroup/

Comment (deads2k) [reply examples/service-catalog/service-catalog-rbac.yaml]: > s/Group/SystemGroup/

That concept disappeared when we collapsed onto upstream.

**Comment by deads2k** [comment]
@pmorie if it works, lgtm

/approve

File: examples/service-catalog/service-catalog-rbac.yaml (hunk 1 above)
Comment (pmorie) [reply examples/service-catalog/service-catalog-rbac.yaml]: Interesting - I didn't see `SystemGroup` as a supported value from the API server when I goofed it at first.

**Comment by pmorie** [comment]
lol.

**Comment by pmorie** [comment]
/retest

<details>

```
Step 1/12 : FROM golang:1.9
Trying to pull repository registry.access.redhat.com/golang ... 
Trying to pull repository docker.io/library/golang ... 
1.9: Pulling from docker.io/library/golang
c73ab1c6897b: Pulling fs layer
1ab373b3deae: Pulling fs layer
b542772b4177: Pulling fs layer
57c8de432dbe: Pulling fs layer
c81227e1ec90: Pulling fs layer
c368dcc533d6: Pulling fs layer
a0037d23043c: Pulling fs layer
f64db6c96f10: Pulling fs layer
c81227e1ec90: Waiting
c368dcc533d6: Waiting
a0037d23043c: Waiting
f64db6c96f10: Waiting
57c8de432dbe: Waiting
b542772b4177: Verifying Checksum
b542772b4177: Download complete
1ab373b3deae: Download complete
c81227e1ec90: Verifying Checksum
c81227e1ec90: Download complete
57c8de432dbe: Verifying Checksum
57c8de432dbe: Download complete
a0037d23043c: Download complete
f64db6c96f10: Verifying Checksum
f64db6c96f10: Download complete
c73ab1c6897b: Verifying Checksum
c73ab1c6897b: Download complete
c368dcc533d6: Verifying Checksum
c368dcc533d6: Download complete
c73ab1c6897b: Pull complete
1ab373b3deae: Pull complete
b542772b4177: Pull complete
57c8de432dbe: Pull complete
c81227e1ec90: Pull complete
c368dcc533d6: Pull complete
a0037d23043c: Pull complete
f64db6c96f10: Pull complete
Digest: sha256:e31bc533c79485cc4ef5a28defd570fa9cafd9257bc86ead8e2e4fa44c5c1397
Status: Downloaded newer image for docker.io/golang:1.9
 ---> 4465fb154d13
Step 2/12 : ENV DEP_VERSION v0.4.1
 ---> Running in 98a654a1ace5
 ---> 5fcfbe9e4081
Removing intermediate container 98a654a1ace5
Step 3/12 : RUN curl -sSL -o /usr/local/bin/dep https://github.com/golang/dep/releases/download/$DEP_VERSION/dep-linux-amd64 &&     chmod +x /usr/local/bin/dep
 ---> Running in c78d24db20ed

 ---> 44bdc983bff0
Removing intermediate container c78d24db20ed
Step 4/12 : RUN curl -sSL https://github.com/coreos/etcd/releases/download/v3.1.10/etcd-v3.1.10-linux-amd64.tar.gz     | tar -vxz -C /usr/local/bin --strip=1 etcd-v3.1.10-linux-amd64/etcd
 ---> Running in 78fedeaccb31

etcd-v3.1.10-linux-amd64/etcd
 ---> c43f1b202796
Removing intermediate container 78fedeaccb31
Step 5/12 : RUN mkdir -p $GOPATH/src/golang.org/x &&     git clone --depth=1 https://github.com/golang/lint.git $GOPATH/src/golang.org/x/lint &&     go get -u golang.org/x/lint/golint
 ---> Running in a8719f9bfa45

Cloning into '/go/src/golang.org/x/lint'...
package golang.org/x/lint/golint: golang.org/x/lint is a custom import path for https://go.googlesource.com/lint, but /go/src/golang.org/x/lint is checked out from https://github.com/golang/lint.git
The command '/bin/sh -c mkdir -p $GOPATH/src/golang.org/x &&     git clone --depth=1 https://github.com/golang/lint.git $GOPATH/src/golang.org/x/lint &&     go get -u golang.org/x/lint/golint' returned a non-zero code: 1
make: *** [.scBuildImage] Error 1
++ export status=FAILURE
++ status=FAILURE
+ set +o xtrace
########## FINISHED STAGE: FAILURE: BUILD SERVICE-CATALOG USER-BROKER IMAGE [00h 01m 16s] ##########
```
</details>

**Comment by jpeeler** [comment]
/retest
The catalog CI failure here should be fixed now.

**Comment by jpeeler** [comment]
/retest
Try again, golint hack was removed.

**Comment by pmorie** [comment]
/retest

👀

**Comment by pmorie** [comment]
Looks very much like the job is busted to me:

```
ansible/playbooks/deploy_cluster.retry'.         [Errno 13] Permission denied:
u'/usr/share/ansible/openshift-ansible/playbooks/deploy_cluster.retry'
localhost                  : ok=82   changed=10   unreachable=0    failed=1   


INSTALLER STATUS ***************************************************************
Initialization             : Complete (0:00:13)
Node Install               : In Progress (0:00:56)
	This phase can be restarted by running: playbooks/openshift-node/config.yml



Failure summary:


  1. Hosts:    localhost
     Play:     Configure nodes
     Task:     Install node, clients, and conntrack packages
     Message:  All items completed
++ export status=FAILURE
++ status=FAILURE
+ set +o xtrace
```

**Pull request merged**